src/config/parser.rs

```diff
@@ -158,18 +119,16 @@ impl Spec {
                     .expect("Internal error: string expected!"),
             );
         }
-        fn probably_ends_spec<I: Iterator<Item = Token>>(iter: &mut Peekable<I>) -> bool {
-            // Check if the iterator "probably ends" here,
-            // or we need to do another round of parsing.
-            ends!(iter)
-                || iter
-                    .peek()
-                    .map(|next| {
-                        [TokType::Semicolon, TokType::MapsTo]
-                            .iter()
-                            .any(|x| next.toktype == *x)
-                    })
-                    .unwrap_or(false)
+        fn starts_spec<I: Iterator<Item = Token>>(iter: &mut Peekable<I>) -> bool {
```

Comment:
This function, `starts_spec()`, is repeatedly used in line `145` and  `160`. To reduce repetition, it may prove useful to rename the function to facilitate returning an `Option` rather than a `bool`.